File: 376.摆动序列.c
/*
 * @lc app=leetcode.cn id=376 lang=c
 *
 * [376] 摆动序列
 */

// @lc code=start

int wiggleMaxLength(int *nums, int numsSize)
{
  int pre, cur, res, i;
  pre = cur = 0;
  res = 1;
  for (i = 0; i < numsSize - 1; i++)
  {
    cur = nums[i + 1] - nums[i];
    if (pre >= 0 && cur < 0 || pre <= 0 && cur > 0)
    {
      res++;
      pre = cur;
    }
  }
  return res;
}
// @lc code=end

File: 376.摆动序列.js
/*
 * @lc app=leetcode.cn id=376 lang=javascript
 *
 * [376] 摆动序列
 */

// @lc code=start
/**
 * @param {number[]} nums
 * @return {number}
 */
var wiggleMaxLength = function (nums) {
  if (nums.length === 1 || (nums.length === 2 && nums[0] === nums[1])) return 1;
  let last,
    i = (num = 0),
    j = 1;
  while (i < nums.length && j < nums.length) {
    const temp = nums[j] - nums[i];
    console.log(last, temp);
    if (last && last * temp < 0) {
      num++;
      i++;
    }
    if (last === undefined) {
      i++;
      num++;
    }
    last = temp;
    j++;
  }
  return num + 1;
};
// @lc code=end
